# DFT+eDMFT

First, we need to optimize the lattice structure using WIEN2k. You can find the generated GaAs structure file in the GitHub repository.

Before running the DMFT code, we need to generate the supercell structure with some atoms displaced from their equilibrium positions.


In [46]:
%%bash
phonopy -d --dim="2 2 2" -c GaAs.struct --wien2k

/home/max/.conda/lib/python3.10/site-packages/spglib/spglib.py:115: DeprecationWarning: dict interface (SpglibDataset['rotations']) is deprecated.Use attribute interface ({self.__class__.__name__}.{key}) instead
  warnings.warn(
/home/max/.conda/lib/python3.10/site-packages/spglib/spglib.py:115: DeprecationWarning: dict interface (SpglibDataset['translations']) is deprecated.Use attribute interface ({self.__class__.__name__}.{key}) instead
  warnings.warn(
/home/max/.conda/lib/python3.10/site-packages/spglib/spglib.py:115: DeprecationWarning: dict interface (SpglibDataset['international']) is deprecated.Use attribute interface ({self.__class__.__name__}.{key}) instead
  warnings.warn(
/home/max/.conda/lib/python3.10/site-packages/spglib/spglib.py:115: DeprecationWarning: dict interface (SpglibDataset['number']) is deprecated.Use attribute interface ({self.__class__.__name__}.{key}) instead
  warnings.warn(
/home/max/.conda/lib/python3.10/site-packages/spglib/spglib.py:115: DeprecationW

        _
  _ __ | |__   ___  _ __   ___   _ __  _   _
 | '_ \| '_ \ / _ \| '_ \ / _ \ | '_ \| | | |
 | |_) | | | | (_) | | | | (_) || |_) | |_| |
 | .__/|_| |_|\___/|_| |_|\___(_) .__/ \__, |
 |_|                            |_|    |___/
                                      2.26.6

Compiled with OpenMP support (max 1 threads).
Python version 3.10.9
Spglib version 2.5.0

Calculator interface: wien2k
Crystal structure was read from "GaAs.struct".
Unit of length: au
Displacements creation mode
Settings:
  Supercell: [2 2 2]
Spacegroup: R3m (160)
Number of symmetry operations in supercell: 48
Use -v option to watch primitive cell, unit cell, and supercell structures.

Number of non-equivalent atoms in GaAs.structS-001.in: 12
Number of non-equivalent atoms in GaAs.structS-002.in: 12
Number of non-equivalent atoms in GaAs.structS-003.in: 12
Number of non-equivalent atoms in GaAs.structS-004.in: 12
"phonopy_disp.yaml" and supercells have been created.

Summary of calculation was written in "

It generates 5 different supercell structures GaAs.structS, GaAs.structS-001, GaAs.structS-002, GaAs.structS-003, FeSe.structS-004. We should run the DMFT calculations for of these structures files. After that we can employ DMFT calculations for each of them.

Now we perform self consistent calculations inside each of the 4 directories, for each of the 4 structures. Here we just show the process for one of them.

In [ ]:
cd GaAs1
init_lapw

<details open>
    <summary><i>Bash Output</i></summary>
<pre><b> next is setrmt </b>
Automatic determination of RMTs. Please specify the desired RMT reduction 
compared to almost touching spheres.
Typically, for a single calculation just hit enter, for force minimization
use 1-5; for volume effects you may need even larger reductions.
 
Enter reduction in %

Use old or new scheme (o/N)

 specify nn-bondlength factor: (usually=2) [and optionally dlimit, dstmax (about 1.d-5, 20)]
 DSTMAX:   24.123128280000003     
 iix,iiy,iiz           2           2           2   43.465096000000003        43.465096000000003        43.465096000000003     

    ATOM  1  Ga         ATOM  9  As        
 RMT(  1)=2.33000 AND RMT(  9)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.69372

    ATOM  2  Ga         ATOM 10  As        
 RMT(  2)=2.33000 AND RMT( 10)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  3  Ga         ATOM 11  As        
 RMT(  3)=2.33000 AND RMT( 11)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  4  Ga         ATOM 12  As        
 RMT(  4)=2.33000 AND RMT( 12)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  5  Ga         ATOM 13  As        
 RMT(  5)=2.33000 AND RMT( 13)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  6  Ga         ATOM 14  As        
 RMT(  6)=2.33000 AND RMT( 14)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  7  Ga         ATOM 15  As        
 RMT(  7)=2.33000 AND RMT( 15)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  8  Ga         ATOM 16  As        
 RMT(  8)=2.33000 AND RMT( 16)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  9  As         ATOM  1  Ga        
 RMT(  9)=2.33000 AND RMT(  1)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.69372

    ATOM 10  As         ATOM  2  Ga        
 RMT( 10)=2.33000 AND RMT(  2)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 11  As         ATOM  3  Ga        
 RMT( 11)=2.33000 AND RMT(  3)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 12  As         ATOM  4  Ga        
 RMT( 12)=2.33000 AND RMT(  4)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 13  As         ATOM  5  Ga        
 RMT( 13)=2.33000 AND RMT(  5)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 14  As         ATOM  6  Ga        
 RMT( 14)=2.33000 AND RMT(  6)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 15  As         ATOM  7  Ga        
 RMT( 15)=2.33000 AND RMT(  7)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 16  As         ATOM  8  Ga        
 RMT( 16)=2.33000 AND RMT(  8)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
  
  NN created a new GaAs1.struct_nn file
STOP NN created a new CASE.STRUCT_NN FILE
0.0u 0.0s 0:00.03 33.3% 0+0k 0+88io 0pf+0w
atom  Z   RMT-max   RMT 
 1  31.0  2.33   2.33  
 2  31.0  2.34   2.33  
 3  31.0  2.34   2.33  
 4  31.0  2.34   2.33  
 5  31.0  2.34   2.33  
 6  31.0  2.34   2.33  
 7  31.0  2.34   2.33  
 8  31.0  2.34   2.33  
 9  33.0  2.33   2.33  
 10  33.0  2.34   2.33  
 11  33.0  2.34   2.33  
 12  33.0  2.34   2.33  
 13  33.0  2.34   2.33  
 14  33.0  2.34   2.33  
 15  33.0  2.34   2.33  
 16  33.0  2.34   2.33  
file    GaAs1.struct_setrmt   generated  
Do you want to accept these radii; discard them; or rerun setRmt (a/d/r):
a
<b> next is nn </b>
&gt;   nn	(15:08:08)  specify nn-bondlength factor: (usually=2) [and optionally dlimit, dstmax (about 1.d-5, 20)]
2
 DSTMAX:   24.123128280000003     
 iix,iiy,iiz           2           2           2   43.465096000000003        43.465096000000003        43.465096000000003     

    ATOM  1  Ga         ATOM  9  As        
 RMT(  1)=2.33000 AND RMT(  9)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.69372

    ATOM  2  Ga         ATOM 10  As        
 RMT(  2)=2.33000 AND RMT( 10)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  3  Ga         ATOM 11  As        
 RMT(  3)=2.33000 AND RMT( 11)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  4  Ga         ATOM 12  As        
 RMT(  4)=2.33000 AND RMT( 12)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  5  Ga         ATOM 13  As        
 RMT(  5)=2.33000 AND RMT( 13)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  6  Ga         ATOM 14  As        
 RMT(  6)=2.33000 AND RMT( 14)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  7  Ga         ATOM 15  As        
 RMT(  7)=2.33000 AND RMT( 15)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  8  Ga         ATOM 16  As        
 RMT(  8)=2.33000 AND RMT( 16)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  9  As         ATOM  1  Ga        
 RMT(  9)=2.33000 AND RMT(  1)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.69372

    ATOM 10  As         ATOM  2  Ga        
 RMT( 10)=2.33000 AND RMT(  2)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 11  As         ATOM  3  Ga        
 RMT( 11)=2.33000 AND RMT(  3)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 12  As         ATOM  4  Ga        
 RMT( 12)=2.33000 AND RMT(  4)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 13  As         ATOM  5  Ga        
 RMT( 13)=2.33000 AND RMT(  5)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 14  As         ATOM  6  Ga        
 RMT( 14)=2.33000 AND RMT(  6)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 15  As         ATOM  7  Ga        
 RMT( 15)=2.33000 AND RMT(  7)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 16  As         ATOM  8  Ga        
 RMT( 16)=2.33000 AND RMT(  8)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
 WARNING: Mult not equal. PLEASE CHECK outputnn-file
 WARNING: ityp not equal. PLEASE CHECK outputnn-file
  
  NN created a new GaAs1.struct_nn file
STOP NN created a new CASE.STRUCT_NN FILE
0.0u 0.0s 0:24.68 0.0% 0+0k 24+88io 0pf+0w
-----&gt; check in  GaAs1.outputnn  for overlapping spheres, 
       coordination and nearest neighbor distances
nn has created a new struct file with different multiplicities
-----&gt; DO YOU WANT TO USE THE NEW GaAs1.struct_nn file (y/n)
y
 Original struct file is saved to GaAs1.struct_init
<b> next is nn </b>
&gt;   nn	(15:09:00)  specify nn-bondlength factor: (usually=2) [and optionally dlimit, dstmax (about 1.d-5, 20)]
2
 DSTMAX:   24.123128280000003     
 iix,iiy,iiz           2           2           2   43.465096000000003        43.465096000000003        43.465096000000003     
 NAMED ATOM: Ga1       Z changed to IATNR+999 to determine equivalency
 NAMED ATOM: Ga2       Z changed to IATNR+999 to determine equivalency
 NAMED ATOM: Ga3       Z changed to IATNR+999 to determine equivalency
 NAMED ATOM: Ga4       Z changed to IATNR+999 to determine equivalency
 NAMED ATOM: Ga5       Z changed to IATNR+999 to determine equivalency
 NAMED ATOM: Ga6       Z changed to IATNR+999 to determine equivalency
 NAMED ATOM: As7       Z changed to IATNR+999 to determine equivalency
 NAMED ATOM: As8       Z changed to IATNR+999 to determine equivalency
 NAMED ATOM: As9       Z changed to IATNR+999 to determine equivalency
 NAMED ATOM: As10      Z changed to IATNR+999 to determine equivalency
 NAMED ATOM: As11      Z changed to IATNR+999 to determine equivalency
 NAMED ATOM: As12      Z changed to IATNR+999 to determine equivalency

    ATOM  1  Ga1        ATOM  7  As7       
 RMT(  1)=2.33000 AND RMT(  7)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.69372

    ATOM  2  Ga2        ATOM  8  As8       
 RMT(  2)=2.33000 AND RMT(  8)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  3  Ga3        ATOM  9  As9       
 RMT(  3)=2.33000 AND RMT(  9)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  4  Ga4        ATOM 10  As10      
 RMT(  4)=2.33000 AND RMT( 10)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  5  Ga5        ATOM 11  As11      
 RMT(  5)=2.33000 AND RMT( 11)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  6  Ga6        ATOM 12  As12      
 RMT(  6)=2.33000 AND RMT( 12)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  7  As7        ATOM  1  Ga1       
 RMT(  7)=2.33000 AND RMT(  1)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.69372

    ATOM  8  As8        ATOM  2  Ga2       
 RMT(  8)=2.33000 AND RMT(  2)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM  9  As9        ATOM  3  Ga3       
 RMT(  9)=2.33000 AND RMT(  3)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 10  As10       ATOM  4  Ga4       
 RMT( 10)=2.33000 AND RMT(  4)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 11  As11       ATOM  5  Ga5       
 RMT( 11)=2.33000 AND RMT(  5)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523

    ATOM 12  As12       ATOM  6  Ga6       
 RMT( 12)=2.33000 AND RMT(  6)=2.33000
 SUMS TO 4.66000  LT.  NN-DIST= 4.70523
STOP NN ENDS
0.0u 0.0s 0:07.76 0.0% 0+0k 16+80io 0pf+0w
-----&gt; check in  GaAs1.outputnn  for overlapping spheres, 
       coordination and nearest neighbor distances
-----&gt; continue with sgroup or edit the GaAs1.struct file (c/e)
c
<b> next is sgroup </b>
&gt;   sgroup	(15:09:08) 0.0u 0.0s 0:00.00 0.0% 0+0k 0+16io 0pf+0w
  Names of point group: m      m      Cs
  Names of point group: m      m      Cs
  Names of point group: 1      1      C1
  Names of point group: 1      1      C1
  Names of point group: m      m      Cs
  Names of point group: m      m      Cs
  Names of point group: m      m      Cs
  Names of point group: m      m      Cs
  Names of point group: 1      1      C1
  Names of point group: 1      1      C1
  Names of point group: m      m      Cs
  Names of point group: m      m      Cs
Number and name of space group: 8 (C m) [unique axis c] cell choice 2
warning: !!! Bravais lattice has changed.
-----&gt; check in  GaAs1.outputsgroup  for proper symmetry, compare
       with your struct file and later with  GaAs1.outputs
       sgroup has also produced a new struct file based on your old one.
       If you see warnings above, consider to use the newly generated 
       struct file, which you can view (edit) now.
-----&gt; continue with symmetry (old case.struct) or use/edit GaAs1.struct_sgroup ? (c/e)
c
<b> next is symmery </b>
&gt;   symmetry	(15:09:16)  SPACE GROUP DOES NOT CONTAIN INVERSION
0.0u 0.0s 0:00.04 0.0% 0+0k 160+64io 1pf+0w
-----&gt; check in  GaAs1.outputs  the symmetry operations, 
       the point symmetries and compare with results from sgroup
       if you find errors (often from rounding errors of positions), apply x patchsymm
-----&gt; continue with lstart or edit the GaAs1.struct_st file (c/e/x)
c
<b> next is lstart </b>
 CREATE A NEW GaAs1.inst FILE with PROPER ATOMS
 If necessary specify switches for instgen_lapw (or press ENTER): 
 -up (default)   -dn   -nm (non-magnetic)   -ask 

 12 Atoms found:  with labels Ga1  Ga2  Ga3  Ga4  Ga5  Ga6  As7  As8  As9  As10 As11 As12 
generate atomic configuration for atom 1 : Ga1
generate atomic configuration for atom 2 : Ga2
generate atomic configuration for atom 3 : Ga3
generate atomic configuration for atom 4 : Ga4
generate atomic configuration for atom 5 : Ga5
generate atomic configuration for atom 6 : Ga6
generate atomic configuration for atom 7 : As7
generate atomic configuration for atom 8 : As8
generate atomic configuration for atom 9 : As9
generate atomic configuration for atom 10 : As10
generate atomic configuration for atom 11 : As11
generate atomic configuration for atom 12 : As12
&gt;   lstart	(15:10:32)   SELECT XCPOT:
  recommended: PBE    [(13) GGA of Perdew-Burke-Ernzerhof 96]
               LDA    [( 5)]
               WC     [(11)  GGA of Wu-Cohen 2006]
               PBESOL [(19) GGA of Perdew etal. 2008]
5
  SELECT ENERGY to separate core and valence states:
  recommended: -6.0 Ry (check how much core charge leaks out of MT-sphere)
  ALTERNATIVELY: specify charge localization (between 0.97 and 1.0) to select core state
-6
STOP LSTART ENDS
1.6u 0.0s 0:22.35 7.8% 0+0k 256+5016io 1pf+0w
-----&gt; continue with kgen or edit the GaAs1.inst file and rerun lstart (c/e)
c
<b> atom 1 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
<b> atom 2 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
<b> atom 3 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
<b> atom 4 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
<b> atom 5 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
<b> atom 6 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
<b> atom 7 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
<b> atom 8 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
<b> atom 9 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
<b> atom 10 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
<b> atom 11 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
<b> atom 12 has a large sphere , consider setting HDLOs and/or larger LVNS </b>
-----&gt; in  GaAs1.in1_st  select   RKmax ( usually 5.0 - 9.0 ), LVNS and LOs
-----&gt; in  GaAs1.in2_st  select   GMAX and Fermi-Energy method
&gt;   inputfiles prepared	(15:11:10)  
<b> inputfiles prepared </b>
<b> inversion is NOT present </b>
&gt;   inputfiles for lapw1c/2c prepared, no inversion present	(15:11:10)  
<b> next is kgen </b>
&gt;   kgen	(15:11:10)            2  symmetry operations without inversion
 inversion added (non-spinpolarized non-so calculation)
  NUMBER OF K-POINTS IN WHOLE CELL: (0 allows to specify 3 divisions of G)
200
 length of reciprocal lattice vectors:   0.289   0.289   0.289   5.848   5.848   5.848
  Shift of k-mesh allowed. Do you want to shift: (0=no, 1=shift)
1
          39  k-points generated, ndiv=           5           5           5
STOP KGEN ENDS
0.0u 0.0s 0:23.43 0.0% 0+0k 96+80io 1pf+0w
-----&gt; check in  GaAs1.klist  number of generated K-points
-----&gt; continue with dstart or execute kgen again or exit (c/e/x)
c
<b> next is dstart </b>
&gt;   dstart -c -p	(15:11:45) running dstart in single mode
STOP DSTART ENDS
60.1u 0.4s 1:02.36 97.1% 0+0k 592+47296io 2pf+0w
-----&gt; check in  GaAs1.outputd  if gmax &gt; gmin, normalization
-----&gt; new GaAs1.in0 generated
-----&gt; do you want to perform a spinpolarized calculation ? (n/y)
n
<b>  init_lapw finished ok  </b>
</pre>
</details>

After that we need to initialize the inputs for DMFT calculations.

In [ ]:
init_dmft.py

<details>
    <summary><i>Bash Output</i></summary>
    <pre>DmftEnvironment: mpi_prefix.dat does not exist -- running in single-processor mode.
There are 16 atoms in the unit cell:
  1 Ga1
  2 Ga2
  3 Ga3
  4 Ga3
  5 Ga4
  6 Ga4
  7 Ga5
  8 Ga6
  9 As7
 10 As8
 11 As9
 12 As9
 13 As10
 14 As10
 15 As11
 16 As12
Specify correlated atoms (ex: 1-4,7,8): 1-8
You have chosen the following atoms to be correlated:
  1 Ga1
  2 Ga2
  3 Ga3
  4 Ga3
  5 Ga4
  6 Ga4
  7 Ga5
  8 Ga6
Do you want to continue; or edit again? (c/e): c

For each atom, specify correlated orbital(s) (ex: d,f):
  1 Ga1: d
  2 Ga2: d
  3 Ga3: d
  4 Ga3: d
  5 Ga4: d
  6 Ga4: d
  7 Ga5: d
  8 Ga6: d
You have chosen to apply correlations to the following orbitals:
  1  Ga1-1 d
  2  Ga2-2 d
  3  Ga3-3 d
  4  Ga3-4 d
  5  Ga4-5 d
  6  Ga4-6 d
  7  Ga5-7 d
  8  Ga6-8 d
Do you want to continue; or edit again? (c/e): c

Specify qsplit for each correlated orbital (default = 0):
    Qsplit  Description
------  ------------------------------------------------------------
     0  average GF, non-correlated
     1  |j,mj&gt; basis, no symmetry, except time reversal (-jz=jz)
    -1  |j,mj&gt; basis, no symmetry, not even time reversal (-jz=jz)
     2  real harmonics basis, no symmetry, except spin (up=dn)
    -2  real harmonics basis, no symmetry, not even spin (up=dn)
     3  t2g orbitals 
    -3  eg orbitals
     4  |j,mj&gt;, only l-1/2 and l+1/2
     5  axial symmetry in real harmonics
     6  hexagonal symmetry in real harmonics
     7  cubic symmetry in real harmonics
     8  axial symmetry in real harmonics, up different than down
     9  hexagonal symmetry in real harmonics, up different than down
    10  cubic symmetry in real harmonics, up different then down
    11  |j,mj&gt; basis, non-zero off diagonal elements
    12  real harmonics, non-zero off diagonal elements
    13  J_eff=1/2 basis for 5d ions, non-magnetic with symmetry
    14  J_eff=1/2 basis for 5d ions, no symmetry
------  ------------------------------------------------------------
  1  Ga1-1 d: 2
  2  Ga2-2 d: 2
  3  Ga3-3 d: 2
  4  Ga3-4 d: 2
  5  Ga4-5 d: 2
  6  Ga4-6 d: 2
  7  Ga5-7 d: 2
  8  Ga6-8 d: 2
You have chosen the following qsplits:
  1  Ga1-1 d: 2
  2  Ga2-2 d: 2
  3  Ga3-3 d: 2
  4  Ga3-4 d: 2
  5  Ga4-5 d: 2
  6  Ga4-6 d: 2
  7  Ga5-7 d: 2
  8  Ga6-8 d: 2
Do you want to continue; or edit again? (c/e): c

Specify projector type (default = 5):
  Projector  Description
------  ------------------------------------------------------------
     1  projection to the solution of Dirac equation (to the head)
     2  projection to the Dirac solution, its energy derivative, 
          LO orbital, as described by P2 in PRB 81, 195107 (2010)
     4  similar to projector-2, but takes fixed number of bands in
          some energy range, even when chemical potential and 
          MT-zero moves (folows band with certain index)
     5  fixed projector, which is written to projectorw.dat. You can
        generate projectorw.dat with the tool wavef.py
     6  similar to projector 5, except projector is here momentum dependent, 
        i.e., similarly to Wannier orbital construction
------  ------------------------------------------------------------
 &gt; 5
5
WARNING: Energy files are not present in this directory. Please generate/copy case.energy files here when using projector 5.

Do you want to continue; or edit again? (c/e): c

Do you want to group any of these orbitals into cluster-DMFT problems? (y/n): n

Enter the correlated problems forming each unique correlated
problem, separated by spaces (ex: 1,3 2,4 5-8): 1-8

Each set of equivalent correlated problems are listed below:
  1   (Ga11 d) (Ga22 d) (Ga33 d) (Ga34 d) (Ga45 d) (Ga46 d) (Ga57 d) (Ga68 d) are equivalent.
Do you want to continue; or edit again? (c/e): c

Range (in eV) of hybridization taken into account in impurity
problems; default -10.0, 10.0: 
-10.0 10.0

Perform calculation on real; or imaginary axis? (i/r): (default=i)i
i

-----&gt; Edit GaAs1.indmf_st file.
-----&gt; Edit GaAs1.indmfl_st file.
</pre>
</details>

Now we are ready to run DMFT.

In [ ]:
run_dmft.py

<details>
    <summary><i>Bash Output</i></summary>
    <pre>Traceback (most recent call last):
  File &quot;/home/max/3EDMFTF/bin/run_dmft.py&quot;, line 19, in &lt;module&gt;
    import oca, ctqmc
  File &quot;/home/max/3EDMFTF/bin/ctqmc.py&quot;, line 11, in &lt;module&gt;
    import dmft_DC
  File &quot;/home/max/3EDMFTF/bin/dmft_DC.py&quot;, line 13, in &lt;module&gt;
    from CoulUs import CoulUsC2_diagonal  # local DMFT Coulomb U matrix, not four but two index matrix
  File &quot;/home/max/3EDMFTF/bin/CoulUs.py&quot;, line 3, in &lt;module&gt;
    import gaunt
ImportError: libifport.so.5: cannot open shared object file: No such file or directory</pre>

    </details>

### References:
[[1]](https://doi.org/10.1103/PhysRevB.81.195107) Kristjan Haule, Chuck-Hou Yee, Kyoo Kim, Phys. Rev. B 81, 195107 (2010).

[[2]](http://hdl.handle.net/20.500.12708/100909) Blaha, Peter & Schwarz, Karlheinz & Madsen, Georg & Kvasnicka, D. & Luitz, J.. (2001). WIEN2k: An Augmented Plane Wave plus Local Orbitals Program for Calculating Crystal Properties. Technische Universität Wien, Wien. 28. 

[[3]](https://doi.org/10.1103/PhysRevB.75.155113) Kristjan Haule, Phys. Rev. B 75, 155113 (2007).

[[4]](https://doi.org/10.1103/PhysRevLett.115.256402) Kristjan Haule, Turan Birol, Phys. Rev. Lett. 115, 256402 (2015).

[[5]](https://doi.org/10.48550/arXiv.1602.02819) Kristjan Haule, Gheorghe L. Pascut, arXiv:1602.02819.


